test/test_tutorial.cpp

```diff
@@ -0,0 +1,173 @@
+  fusion.addInput(tv0);
+
+  // Just create a copy
+  TensorView* tv1 = set(tv0);
+  fusion.addOutput(tv1);
+
+  // End of fusion definition
+
+  if (verbose_) {
+    // Here's some common ways to inspect the fusion. These are not
+    // necessary for running the fusion but should provide helpful
+    // information for understanding how fusions are transformed.
+
+    // Print a concise representation of the fusion exprssions
+    fusion.printMath();
+
+    // Generate and print a CUDA kernel. Notice that at this point the
+    // genereated code is just a sequential kernel as we have not
+    // scheduled the fusion yet, but it should be a valid CUDA kernel
+    fusion.printKernel();
+  }
+
+  // Next, try running the fusion. First, we need to set up a sample
+  // input tensor. Here, we create a 32x32 tensor initialized with
+  // random float values.
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({32, 32}, options);
+  std::vector<c10::IValue> aten_inputs = {t0};
+
+  // Next, lower the fusion to Kernel, generate CUDA kernel source and then
+  // compile it with nvrtc. All of them are done by FusionExecutor
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+
+  // FusionExecutor now has a compiled kernel, which can be executed as:
+  std::vector<at::Tensor> outputs = fe.runFusion(aten_inputs);
+  // Note that this run is done using just one thread, which will be
+  // corrected below.
+
+  // To validate the output, we can just assert that the output is
+  // equal to the input as this is just a copy fusion. More commonly,
+  // though, testValidate is used to validate outputs while
+  // automatically adjusting thresholds of valid deviations
+  ASSERT_TRUE(outputs[0].equal(t0));
+
+  // Next, instead of just running the fusion as is, we manually
+  // schedule it so that it runs in parallel. In this case, we only
+  // have one expression, i.e., the set expression, so we just need to
+  // schedule tv1.
+
+  // tv1 is a 2D tensor. Let its domain be [i0, i1]. What we are going
+  // to do is to transform this 2D domain to the multi-dimensional
+  // CUDA parallelism, i.e., a grid consisting of multiple thread
+  // blocks, each of which consisting of multiple threads. A common
+  // transformation pattern is to merge all of each axis to get a
+  // flattened domain, and then split the domain to factor out axes
+  // that are parallelized by threads and thread blocks.
+
+  // For example, the current domain of tv1 looks like [i0, i1]. We
+  // can merge the two axes by:
+  tv1->merge(0, 1);
+  // This creates a single axis that merges i0 and i1. Its extent is a
+  // multiplication of the extents of i0 and i1, so we commonly
+  // represent it as [i0*i1]. It can be also examined with:
+  if (verbose_) {
+    std::cout << tv1->toString() << std::endl;
+  }
+
+  // Next, we factor out a subdomain for threads in each thread
+  // block.
+  tv1->split(0, 256);
```

Comment:
Is this 256 arbitrarily chosen or mapping to some hardware feature?

Reply:
Good question. Try running the test with `NVFUSER_DUMP=launch_param`. You'll see the kernel is executed with 256 thread per thread block.

Also see this comment if it answers your question: https://github.com/NVIDIA/Fuser/pull/854/files/63c2e9cf6cabbefe4402ae943b2672b84e042277#diff-456c309c5c47db3d134fc0cad9b8efe9eddda10a5d8a88dd79ff5b0bc666ed7dR140-R145

Reply:
Got it, thank you for the answer!

Reply:
Sorry a basic question, what is the command to just run this one test?

Reply:
This should work:

` ./bin/nvfuser_tests --gtest_filter="Tutorial.Memcpy"`

Reply:
Cool, saw the Launch Parameters: 
Launch Parameters: BlockDim.x = -1, BlockDim.y = -1, BlockDim.z = -1, GridDim.x = -1, GridDim.y = -1, GridDim.z = -1, Smem Size = 0
**Launch Parameters: BlockDim.x = 256**, BlockDim.y = -1, BlockDim.z = -1, GridDim.x = 4, GridDim.y = -1, GridDim.z = -1, Smem Size = 0

Reply:
> Sorry a basic question, what is the command to just run this one test?
> ./bin/nvfuser_tests --gtest_filter="Tutorial.Memcpy"

This can be added to the code comment.